src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+        } else {
+            return CompletableFuture.completedFuture(null);
+        }
+    }
+
+    /**
+     * Fetches existing {@link InventoryEntry} objects from CTP project that correspond to passed {@code batchOfDrafts}.
+     * Having existing inventory entries fetched, {@code batchOfDrafts} is compared and synced with fetched objects by
+     * {@link InventorySync#compareAndSync(Map, List)} function. When fetching existing inventory entries results in
+     * exception then error callback is executed and {@code batchOfDrafts} isn't processed.
+
+     * @param batchOfDrafts batch of drafts that need to be synced
+     * @return {@link CompletionStage} of {@link Void} that indicates method progress.
+     */
+    private CompletionStage<Void> processBatch(final List<InventoryEntryDraft> batchOfDrafts) {
+        return fetchExistingInventories(batchOfDrafts)
+                .thenCompose(existingInventories -> compareAndSync(existingInventories, batchOfDrafts))
+                .exceptionally(ex -> {
+                    handleFailure(format("Failed to find existing inventory entries of SKUs %s",
+                            extractSkus(batchOfDrafts)), ex);
+                    return null;
+                });
+    }
+
+    /**
+     * For each draft from {@code drafts} it checks if there is corresponding entry in {@code existingInventories},
+     * and then either attempts to update such entry with data from draft or attempts to create new entry from draft.
+     * After comparision and performing action "processed" counter from statistics is incremented.
+     * Method returns {@link CompletionStage} of {@link Void} that indicates all possible creation/update attempts
+     * progress.
+     *
+     * @param existingInventories mapping of {@link SkuKeyTuple} to {@link InventoryEntry} of instances existing in a
+     *                            CTP project
+     * @param drafts drafts that need to be synced
+     * @return {@link CompletionStage} of {@link Void} that indicates all possible creation/update attempts progress.
+     */
+    private CompletionStage<Void> compareAndSync(final Map<SkuKeyTuple, InventoryEntry> existingInventories,
+                                                 final List<InventoryEntryDraft> drafts) {
+        final List<CompletableFuture<Void>> futures = new ArrayList<>(drafts.size());
+        for (InventoryEntryDraft draft : drafts) {
+            final SkuKeyTuple skuKeyOfDraft = SkuKeyTuple.of(draft);
+            if (existingInventories.containsKey(skuKeyOfDraft)) {
+                final InventoryEntry existingEntry = existingInventories.get(skuKeyOfDraft);
+                futures.add(attemptUpdate(existingEntry, draft).toCompletableFuture());
+            } else {
+                futures.add(attemptCreate(draft).toCompletableFuture());
+            }
+            statistics.incrementProcessed();
```

Comment:
There is an issue for statistics https://github.com/commercetools/commercetools-sync-java/issues/23 so it will be handled in another branch/PR